backend/src/Tabulators/IRV.ts

```diff
@@ -0,0 +1,241 @@
+        summaryData: summaryData,
+        logs: [],
+        voteCounts: [],
+        exhaustedVoteCounts: [],
+        overVoteCounts: []
+    }
+
+    let remainingCandidates = [...summaryData.candidates]
+    let activeVotes = parsedData.scores
+    let exhaustedVotes: ballot[] = []
+    let exhaustedVoteCount = 0
+    let overVoteCount = 0
+    while (results.elected.length < 1) { //TODO: multiwinner STV
+        let roundVoteCounts = Array(summaryData.candidates.length).fill(0)
+        for (let i = activeVotes.length - 1; i >= 0; i--) {
+            // loop backwards over the ballots so exhausted votes can be removed
+            let ballot = activeVotes[i]
+            let topRemainingRank: number = 0
+            let topRemainingRankIndex: number = 0
+            let isOverVote = false
+            remainingCandidates.forEach(candidate => {
+                // loop over remaining candidates
+                // find top (non-zero) rank, if duplicate top ranks mark as overvote
+                if (ballot[candidate.index] !== 0) {
+                    // candidate has rank
+
+                    if (topRemainingRank == 0) {
+                        // set initial top rank
+                        topRemainingRankIndex = candidate.index
+                        topRemainingRank = ballot[candidate.index]
+                        isOverVote = false
+                    }
+                    else if (ballot[candidate.index] < ballot[topRemainingRankIndex]) {
+                        // set new top rank
+                        topRemainingRankIndex = candidate.index
+                        topRemainingRank = ballot[candidate.index]
+                        isOverVote = false
+                    }
+                    else if (ballot[candidate.index] === ballot[topRemainingRankIndex]) {
+                        // multiple top ranks, mark as over vote
+                        isOverVote = true
+                    }
+                }
+            })
+            if (isOverVote) {
+                // ballot is overvote
+                overVoteCount += 1
+            }
+            if (topRemainingRank === 0 || isOverVote) {
+                // ballot is exhausted
+                exhaustedVoteCount += 1
+                exhaustedVotes.push(...activeVotes.splice(i, 1))
+            }
+            else {
+                // give vote to top rank candidate
+                roundVoteCounts[topRemainingRankIndex] += 1
+            }
+        }
+        results.voteCounts.push(roundVoteCounts)
+        results.exhaustedVoteCounts.push(exhaustedVoteCount)
+        results.overVoteCounts.push(overVoteCount)
+        let voteThreshold = activeVotes.length / 2
+        
+        // get max number of votes
+        let remainingCandidatesIndexes = remainingCandidates.map(c => c.index)
+        let maxVotes = Math.max(...roundVoteCounts.filter((c, i) => remainingCandidatesIndexes.includes(i)))
+        if (maxVotes >= voteThreshold) {
```

Comment:
I don't think this accounts for tie situations. Here's an example

```
A: 50
B: 30
C: 20
```

Here A has >= voteThreshold, but the remaining votes still need to be distributed to determine if A gets a win or if it's a tie between A and B

Reply:
Is the threshold 50%+1?

Reply:
I think so, that makes the most sense to me

This source says 50+1, https://ballotpedia.org/Ranked-choice_voting_(RCV) but I'm not sure if that's the consensus

Reply:
Changed to needing to be greater than 50% threshold, also now selects the candidate with the most votes if your have 2 or less candidates which will handle ties.